openghg/objectstore/metastore/_metastore.py

```diff
@@ -176,7 +181,45 @@ def _get_query(self, metadata: MetaData) -> tinydb.queries.QueryInstance:
         """
         return tinydb.Query().fragment(self._format_metadata(metadata))
 
-    def search(self, search_terms: Optional[MetaData] = None) -> QueryResults:
+    def _get_test_query(self, search_tests: dict[str, Callable]) -> tinydb.queries.QueryInstance:
```

Comment:
`search_tests` replaced with `search_ranges`. For uniformity and code readability. Although the user won't see it.

Reply:
I agree perhaps this could have a more descriptive name but doesn't this input allow for more generic cases than just ranges though? In principle this could be any function so perhaps `search_by_function` (unless that seems too wordy)? The original `search_tests` is also fine by me though if we can't think of an alternative.

Reply:
I changed it to "search tests" because the that is the name of the method used by the TinyDB Query: `Query().test(test_function)`, but maybe `search_by_function` is better.